### PR TITLE
Build with JDK-11 in all Travis jobs

### PR DESCRIPTION
**Pull request by JaroslavTulach**
NetBeans are supposed to be developed with JDK-11 yet nobody updated Travis jobs to do so yet. Adding `nbbuild/travis/ant.sh` script which downloads JDK-11. Using it for compilation, but not for testing. Testing continues to be performed on the JDK the job has been setup for.

By merging this PR we shall be ready to start using [alternative modular implementations](http://wiki.apidesign.org/wiki/AlternativeImplementation). E.g. introduce `org.netbeans.modules.xyz.jdk11` with `javac.target=11` providing better integration with JDK-11 where needed.

### CODE REVIEW

**Comment by neilcsmith-net** [comment]
Can we not consolidate on one method of downloading a JDK in this?  `install-jdk.sh` should also be in the path and not need downloading itself, although that also may not be the best option.  _A_ JDK 11 should also be installed already too.

**Comment by JaroslavTulach** [comment]
> Can we not consolidate on one method of downloading a JDK in this? 

If one trusts that the URL remains persistent, downloading from a URL seems acceptable to me. I have decided to trust this JDK provider in my projects. Moreover this primitive technique is portable across CI types - it shall work everywhere `curl` is installed.

> `install-jdk.sh` should also be in the path and not need downloading itself, although that also may not be the best option. 

I've never used it myself. Neither inside Travis, neither locally (e.g. I cannot test the behavior). The PR is however open for edits from maintainers, if users of `install-jdk.sh` want to step in.

> _A_ JDK 11 should also be installed already too.

I wanted to avoid modifying the behavior of the Travis scripts. For example by accidentally switching to JDK11 for testing. Downloading a JDK into temporary directory and using it only in `ant.sh` guarantees no other part of the scripts is going to be influenced.

**Comment by neilcsmith-net** [comment]
It wasn't an argument for or against `install-jdk.sh`.  It was an argument for picking _one_ way to do this.  We're already using it - eg. https://github.com/apache/netbeans/blob/master/.travis.yml#L445

> I wanted to avoid modifying the behavior of the Travis scripts. For example by accidentally switching to JDK11 for testing.

Understand argument for known JDK.  But don't see why there's any chance of accidentally switching from using `JAVA_HOME` with the already installed JDKs vs one downloaded.  Either way, had wondered merit of running `ant -diagnostics` on each switch of JDK for sanity checking?

**Comment by mbien** [comment]
I added a few echos to the build to make it easier to quickly check which JDK is in path. It prints it very early so that it is easily to find.
https://github.com/apache/netbeans/pull/3754/files#diff-ef48f302a1dc02030dd4f72d068dd2c51d8c4e27fab06014cd0f7a1ae7cd2fa9R41-R43

**Comment by JaroslavTulach** [comment]
I've noticed earlier that [NetBeans 13 has broken profiler when running on JDK8](https://app.travis-ci.com/github/apache/netbeans/jobs/562936591#L20558). The [933ab50](https://github.com/apache/netbeans/pull/3744/commits/933ab50188b85fa1caefb170f2da340289bb267c) commit shall fix that. Sorry for not catching it earlier.

**Comment by mbien** [comment]
wondering if we can remove the "Compile with JDK-11 and test something on JDK-8 " gh job if this PR makes it in.

I also have to agree with Neil, we should not use two different scripts for downloading JDKs if there is a reasonable way to avoid it. install-jdk.sh should be able to install JDKs side by side I believe.

**Comment by JaroslavTulach** [comment]
Looks like [we are green](https://app.travis-ci.com/github/apache/netbeans/jobs/563149156). I squash the changes, accommodate #3778 and then we can review and merge.

**Review by sdedic — APPROVED** [review]

File: nbbuild/travis/ant.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    # download
+    install
+    cd travisbuildjdk
+    echo Java is ready in `pwd`
+    ./bin/java -version
+fi
+cd "$AT"
+
+echo Running ant -q $* at `pwd`
+JAVA_HOME=`pwd`/nbbuild/travisbuildjdk/ ant -q $*
```
[the remark this answers is not in the document]
Comment (JaroslavTulach) [reply nbbuild/travis/ant.sh:60]: I am trying to make the use of the downloaded JDK "self-contained". The goal is to limit influence on the rest of the `.travis.yml` operations. True, changing the `PATH` should not affect the "self-containment". The change would only be visible inside the `ant.sh` script.

But the gates are running fine with `JAVA_HOME` - there doesn't seem to be any need to mangle `PATH`.

File: nbbuild/travis/ant.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+function install() {
+    mkdir -p nbbuild/
+    cd nbbuild/
+    rm -rf travisbuildjdk*
+    wget https://raw.githubusercontent.com/sormuras/bach/master/install-jdk.sh
+    chmod a+x install-jdk.sh
+    ./install-jdk.sh --feature 11 --license GPL --target travisbuildjdk
+}
+
+function download() {
```
[the remark this answers is not in the document]
Comment (JaroslavTulach) [reply nbbuild/travis/ant.sh:31]: There was a discussion whether using `install-jdk.sh` is better than manual download from a fixed URL. I changed the code to use `install-jdk.sh`, but I still prefer the _fixed URL approach_ a bit more:
- `install-jdk.sh` script may (and should, that's the whole point) change over time - e.g. the download may not always be the same
- fixed URL (plus checksum) is really _fixed_. If you trust the URL provider (I trust Azul to keep it unchanged)...
- anyway for purposes of running CI before integration the difference isn't huge either way

 As such I have switched to `install-jdk.sh` but left the original `download` function in the code, not called right now.

File: .travis.yml [hunk 3]
```diff
@@ -76,29 +76,30 @@ matrix:
...
+            - nbbuild/travis/ant.sh -quiet build-source-config -Dcluster.config=platform
             - mkdir tmpplatform
             - cd tmpplatform
             - unzip -qq ../nbbuild/build/platform-src*
-            - ant build -Dcluster.config=platform
+            - chmod a+x nbbuild/travis/ant.sh
+            - nbbuild/travis/ant.sh build -Dcluster.config=platform
         
-        - name: Compile all modules with OpenJDK8
+        - name: Compile all modules & tests
```
Comment (JaroslavTulach) [inline .travis.yml:86]: Now the job compiles the code and the tests with JDK-11.

**Comment by JaroslavTulach** [comment]
We are green. Given how hard it is to get green status from all our CI builders, I'd rather merge than do some other changes (like _remove the "Compile with JDK-11 and test something on JDK-8 " job_) which I've forgotten - they can be done any time later.

By merging this PR we shall be ready to start using [alternative modular implementations](http://wiki.apidesign.org/wiki/AlternativeImplementation). E.g. introduce `org.netbeans.modules.xyz.jdk11` with `javac.target=11` providing better integration with JDK-11 where needed.

**Pull request merged**